"use strict";(self.webpackChunkdocs=self.webpackChunkdocs||[]).push([[53],{1109:a=>{a.exports=JSON.parse('{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"docsSidebar":[{"type":"link","label":"Introduction","href":"/intro","docId":"intro"},{"type":"category","label":"Visualizations","collapsible":true,"collapsed":false,"items":[{"type":"link","label":"Cohort View","href":"/visualizations/cohort-view","docId":"visualizations/cohort-view"},{"type":"link","label":"Genome View","href":"/visualizations/genome-view","docId":"visualizations/genome-view"},{"type":"link","label":"Variant View","href":"/visualizations/variant-view","docId":"visualizations/variant-view"},{"type":"link","label":"Breakpoint View","href":"/visualizations/breakpoint-view","docId":"visualizations/breakpoint-view"},{"type":"link","label":"Navigation","href":"/visualizations/navigation","docId":"visualizations/navigation"},{"type":"link","label":"Export","href":"/visualizations/export","docId":"visualizations/export"},{"type":"link","label":"Data Sampling","href":"/visualizations/data-sampling","docId":"visualizations/data-sampling"}],"href":"/category/visualizations"},{"type":"category","label":"Loading Data","collapsible":true,"collapsed":false,"items":[{"type":"link","label":"Through Interface","href":"/loading-data/through-interface","docId":"loading-data/through-interface"},{"type":"link","label":"Through Data Config","href":"/loading-data/through-data-config","docId":"loading-data/through-data-config"},{"type":"link","label":"Data Formats","href":"/loading-data/data-formats","docId":"loading-data/data-formats"},{"type":"link","label":"Thumbnails","href":"/loading-data/thumbnails","docId":"loading-data/thumbnails"},{"type":"link","label":"URL Parameters","href":"/loading-data/url-parameters","docId":"loading-data/url-parameters"},{"type":"link","label":"Loading Local Data","href":"/loading-data/local-data","docId":"loading-data/local-data"},{"type":"link","label":"Loading Private Data","href":"/loading-data/loading-private-data","docId":"loading-data/loading-private-data"},{"type":"link","label":"Cohort Config Creation","href":"/loading-data/cohort-config-creation","docId":"loading-data/cohort-config-creation"},{"type":"link","label":"Using Python Package","href":"/loading-data/python-package","docId":"loading-data/python-package"}],"href":"/category/loading-data"},{"type":"category","label":"Available Data","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"PCAWG Data","href":"/available-data/pcawg","docId":"available-data/pcawg"},{"type":"link","label":"Cell Line Data with Read-level View","href":"/available-data/cgpwgs","docId":"available-data/cgpwgs"}],"href":"/category/available-data"},{"type":"category","label":"Use Cases","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Global Pattern","href":"/use-cases/global-pattern","docId":"use-cases/global-pattern"},{"type":"link","label":"Biallelic Loss of Tumor suppressor","href":"/use-cases/bi-allelic-loss","docId":"use-cases/bi-allelic-loss"},{"type":"link","label":"Pathogenic BRCA Mutations","href":"/use-cases/brca","docId":"use-cases/brca"},{"type":"link","label":"Co-amplifications","href":"/use-cases/co-amplifications","docId":"use-cases/co-amplifications"},{"type":"link","label":"Biallelic ATM Mutations","href":"/use-cases/atm-mutations","docId":"use-cases/atm-mutations"},{"type":"link","label":"CCNE1 Amplifications","href":"/use-cases/ccne1","docId":"use-cases/ccne1"},{"type":"link","label":"Biallelic Loss of CDK12","href":"/use-cases/cdk12","docId":"use-cases/cdk12"}],"href":"/category/use-cases"},{"type":"link","label":"Workflows","href":"/workflows","docId":"workflows"},{"type":"category","label":"Help","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Q & A","href":"/help/faq","docId":"help/faq"},{"type":"link","label":"Contact","href":"/help/contact","docId":"help/contact"}],"href":"/category/help"}]},"docs":{"available-data/cgpwgs":{"id":"available-data/cgpwgs","title":"Cell Line Data with Read-level View","description":"1. Cell line whole genome data with sequencing reads (2 samples)","sidebar":"docsSidebar"},"available-data/pcawg":{"id":"available-data/pcawg","title":"PCAWG Data","description":"We offer a total of 2,778 samples of 39 cancer types from the Pan-Cancer Analysis of Whole Genomes (PCAWG) consortium.","sidebar":"docsSidebar"},"help/contact":{"id":"help/contact","title":"Contact","description":"If you are running into any issues:","sidebar":"docsSidebar"},"help/faq":{"id":"help/faq","title":"Q & A","description":"Q. Which browsers are supported?","sidebar":"docsSidebar"},"intro":{"id":"intro","title":"Introduction","description":"Welcome to the documentation of Chromoscope!","sidebar":"docsSidebar"},"loading-data/cohort-config-creation":{"id":"loading-data/cohort-config-creation","title":"Cohort Config Creation","description":"Overview","sidebar":"docsSidebar"},"loading-data/data-formats":{"id":"loading-data/data-formats","title":"Data Formats","description":"This page describes file formats used in Chromoscope. To find a list of required and optional files, please refer to the Data Configuration section.","sidebar":"docsSidebar"},"loading-data/loading-private-data":{"id":"loading-data/loading-private-data","title":"Loading Private Data","description":"This article details how to set up security credentials for an Amazon Web Services (AWS) account and temporarily visualize data stored in private S3 buckets (all public access blocked) via presigned URLs. Presigned URLs are used in provided scripts described in the following section, to create configuration files for large cohorts of sample data saved on private S3 buckets.","sidebar":"docsSidebar"},"loading-data/local-data":{"id":"loading-data/local-data","title":"Loading Local Data","description":"You may also want to see the Chromoscope Python package, which allows you to load local files on computational notebooks.","sidebar":"docsSidebar"},"loading-data/python-package":{"id":"loading-data/python-package","title":"Using Python Package","description":"You can use a Chromoscope Python package to visualize your local SV files directly on computational notebooks, such as Jupyter Notebook, Jupyter Lab, and Google Colab.","sidebar":"docsSidebar"},"loading-data/through-data-config":{"id":"loading-data/through-data-config","title":"Through Data Config","description":"You can load a large number of samples through data configurations. You need to (1) make a data config file (.json) that contains the information for individual samples, (2) store the config file in a HTTPS file server (e.g., AWS S3 or GitHub Gist), and (3) use it with the external parameter of the Chromoscope URL:","sidebar":"docsSidebar"},"loading-data/through-interface":{"id":"loading-data/through-interface","title":"Through Interface","description":"You can load a small number of samples directly on the browser through the interface. In the cohort-level view, there is a side panel that enables you to add a new sample by providing metadata (e.g., cancer type) and file URLs (e.g., bedpe, txt, vcf).","sidebar":"docsSidebar"},"loading-data/thumbnails":{"id":"loading-data/thumbnails","title":"Thumbnails","description":"To enable browsing a large number of samples at scale in the cohort-level view, we generated thumbnail images for individual samples and use these image files. We pre-generated thumbnails for all datasets that we provide.","sidebar":"docsSidebar"},"loading-data/url-parameters":{"id":"loading-data/url-parameters","title":"URL Parameters","description":"There are multiple parameters available that you can used along with the base URL (https","sidebar":"docsSidebar"},"use-cases/atm-mutations":{"id":"use-cases/atm-mutations","title":"Biallelic ATM Mutations","description":"The ataxia-telangiectasia mutated (ATM) gene is essential in the detection and signaling to repair DNA double-strand breaks. Carriers of germline mutations in ATM are predisposed to cancer, and ATM is also often mutated somatically. In light of new cancer therapies targeting cells with ATM loss, we hypothesized that ATM loss would result in a mutational signature apparent in cancer genomes.","sidebar":"docsSidebar"},"use-cases/bi-allelic-loss":{"id":"use-cases/bi-allelic-loss","title":"Biallelic Loss of Tumor suppressor","description":"Chromoscope facilitates the interpretation of clinically-relevant SVs in cancer genomes.","sidebar":"docsSidebar"},"use-cases/brca":{"id":"use-cases/brca","title":"Pathogenic BRCA Mutations","description":"The BRCA1 and BRCA2 genes have a role in the accurate repair of DNA double-strand breaks through homologous recombination deficiency. A mutational signature of BRCA1/2 loss consists of multiple mutation types (point mutations, indels, structural and copy number variants). The exact mechanisms of how such mutations form in the context of homologous recombination deficiency have not been fully understood.","sidebar":"docsSidebar"},"use-cases/ccne1":{"id":"use-cases/ccne1","title":"CCNE1 Amplifications","description":"The CCNE1 gene is commonly amplified in different cancer types, and results in oncogenic stress. Cancers with CCNE1 amplifications often display a characteristic signature of structural variants and are dominated by somatic tandem duplications.","sidebar":"docsSidebar"},"use-cases/cdk12":{"id":"use-cases/cdk12","title":"Biallelic Loss of CDK12","description":"The CDK12 gene is a cyclin-dependent kinase and controls the cell cycle, cell division, and transcription.  Cancers with pathogenic CDK12 mutations display chromosomal instability and their genomes are dominated by somatic tandem duplications. The mechanistic connection between the CDK12 loss and chromosomal instability remains unresolved, and uncovering the mechanism might inform new cancer therapies.","sidebar":"docsSidebar"},"use-cases/co-amplifications":{"id":"use-cases/co-amplifications","title":"Co-amplifications","description":"The amplifications of chromosomes 8 and 11 in breast cancer co-occur. An inspection of these loci with Chromoscope shows that the amplifications are often connected by inter-chromosomal translocations, implicating the derivative chromosome 8/11 as an early event in their formation. The segments of loss-of-heterozygosity (LOH) adjacent to the amplicon loci are consistent with this model of amplicon origin. The visualization highlights commonalities in features of the amplicons across 23 samples, as featured in a recent study on the origin of these amplicons.","sidebar":"docsSidebar"},"use-cases/global-pattern":{"id":"use-cases/global-pattern","title":"Global Pattern","description":"Genome-wide occurrences of specific structural variants (SVs) may indicate possible DNA repair deficiencies in cancer. For example, inactivation of a homologous recombination gene, such as BRCA1 and BRCA2, leads to chromosomal instability, thereby increasing the likelihood of therapeutic response to PARP inhibition. In Chromoscope, the genomic signature is apparent as hundreds of scattered deletions and duplications with sizes of 1\u201310 kb are shown in the genome (Fig. 1, left column) and variant views (Fig. 1, right column). In the variant view, the footprint on the copy number profiles is consistent with losses and gains caused by deletions and tandem duplications.","sidebar":"docsSidebar"},"visualizations/breakpoint-view":{"id":"visualizations/breakpoint-view","title":"Breakpoint View","description":"The breakpoint view shows reads around breakpoints and highlights pairs of reads with long distances, showing evidence for structural variant calls.","sidebar":"docsSidebar"},"visualizations/cohort-view":{"id":"visualizations/cohort-view","title":"Cohort View","description":"This cohort-level view displays structural variants as arches connecting chromosomal breakpoints using distinct colors to indicate duplications, deletions, inversions, and inter-chromosomal translocations. This view also displays copy number variants (CNV) and loss of heterozygosity (LOH).","sidebar":"docsSidebar"},"visualizations/data-sampling":{"id":"visualizations/data-sampling","title":"Data Sampling","description":"For the efficient rendering of visualizations, entire data is not shown in the whole genome scale for point mutations, indels, and gene annotations.","sidebar":"docsSidebar"},"visualizations/export":{"id":"visualizations/export","title":"Export","description":"PNG","sidebar":"docsSidebar"},"visualizations/genome-view":{"id":"visualizations/genome-view","title":"Genome View","description":"The genome view shows the selected sample in a circular visualization. This uses the visual representations that are similar to the ones on the cohort view but shows several additional tracks, such as chromosome ideograms and putative drivers.","sidebar":"docsSidebar"},"visualizations/navigation":{"id":"visualizations/navigation","title":"Navigation","description":"All visualizations in Chromoscope are interactive which is designed to support easy-navigation between genomic regions of interest across scales.","sidebar":"docsSidebar"},"visualizations/variant-view":{"id":"visualizations/variant-view","title":"Variant View","description":"The variant view, focusing on a shorter genomic region, shows additional tracks on top of the tracks included in the genome view, including point mutations, indels, copy number variation like gains and loss of heterozgosity (LOH), and genes.","sidebar":"docsSidebar"},"workflows":{"id":"workflows","title":"Workflows","description":"There are several ways to use Chromoscope visualizations in different analysis workflows other than directly using the online instance, https://chromoscope.bio/app/. One way is to use the exported Gosling specification and the other is to use Chromoscope Python Package.","sidebar":"docsSidebar"}}}')}}]);